### PR TITLE
[werft] Cleanup build.ts around KUBECONFIG

### PR DESCRIPTION
**Pull request by geropl**
## Description
Makes build.ts more readable and extensible.

## Related Issue(s)

Fixes #

## How to test

 - see that the deployment works: https://werft.gitpod-dev.com/job/gitpod-build-gpl-dev-werft-chores.2
 - see that the `wipe-devstaging` job works as well: https://werft.gitpod-dev.com/job/gitpod-wipe-devstaging-gpl-dev-werft-chores.1/logs

## Release Notes

```release-note
NONE
```

## Documentation

- [x] /werft k3s-ws

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** after the PR has been reviewed.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by geropl** [comment]
/hold

waiting for other PRs to get merged beforehand

**Comment by princerachit** [comment]
There are some conflicts. Please request another review when they are resolved. And thanks for doing this!

File: .werft/build.ts
Comment (princerachit) [inline .werft/build.ts:734]: Any reason for changing this?

**Comment by princerachit** [comment]
I see this is failing with k3s:
```
NAME        URL                           STATIC        STATE            SCORE        GOVERNED        ADMISSION CONSTRAINTS        ADMISSION PREFERNCES
dev         dns:///ws-manager:8080        true          AVAILABLE        50           true            [][]
[register k3s ws cluster] NAME        URL                           STATIC        STATE            SCORE        GOVERNED        ADMISSION CONSTRAINTS        ADMISSION PREFERNCES
dev         dns:///ws-manager:8080        true          AVAILABLE        50           true            [][]
wrote ws-manager TLS config to: ./wsman-tls
[register k3s ws cluster] wrote ws-manager TLS config to: ./wsman-tls
time="2021-11-09T05:05:15Z" level=fatal msg="rpc error: code = Unknown desc = cannot reach ws-k3s.gpl-dev-werft-chores.staging.gitpod-dev.com:8081: 14 UNAVAILABLE: No connection established"
[register k3s ws cluster] 
time="2021-11-09T05:05:15Z" level=fatal msg="rpc error: code = Unknown desc = a WorkspaceCluster with name k3s already exists in the DB!"
[register k3s ws cluster] 
[register k3s ws cluster|PHASE] done
[trigger integration tests|PHASE] Trigger integration tests
[trigger integration tests] Skipped integration tests
[trigger integration tests|DONE]
About to exit with code 0. Shutting down tracing.
Tracing terminated
About to exit with code 0. Shutting down tracing.
```
Can you TAL if your changes are causing this? Other wise this looks good to me

**Comment by geropl** [comment]
> Can you TAL if your changes are causing this? Other wise this looks good to me

I don't think this is caused by the changes. If you double check [here](https://github.com/gitpod-io/gitpod/blob/7695dcc9fd0c7e3d037633fb162fb131ca277f88/.werft/build.ts#L708-L728) it looks semantical identical to the version before. Not sure what causes this! :shrug:

**Review by princerachit — APPROVED** [review]

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: f6573a90bad62ca982e3c38c7ecdddd19dda0c54</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6503#pullrequestreview-801641595" title="Approved">princerachit</a>*

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [princerachit]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6503#issuecomment-964403795" title="Approved">princerachit</a>*

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [princerachit]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6503#issuecomment-964404506" title="Approved">princerachit</a>*

Associated issue requirement bypassed by: *<a href="https://github.com/gitpod-io/gitpod/pull/6503#issuecomment-964404506" title="Approved">princerachit</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [princerachit]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
